### DISCUSSION TITLE
Problems with BetterBibLaTex accessDate variable

### DISCUSSION THREAD

**Discussion by KevinYanesG** [retorquere/zotero-better-bibtex#2452] [Q&A] [answered]
Hi,
I'm syncing a .bib file from zotero with BibLaTex. When I export web pages I want them to be accompanied with the accesed Date as a note like proposed here: https://retorque.re/zotero-better-bibtex/exporting/scripting/#add-accessdate-url-for-bibtex

This doesn't work for me. The only thing that works is the following:

```javascript
if (Translator.BetterBibLaTeX && zotero.itemType === 'webpage') {
      tex.add({ name: 'note', value: "(accessed " + zotero.accessDate + ")" });
        if (zotero.url) {
      tex.add({ name: 'howpublished', bibtex: "{\\url{" + tex.enc_verbatim({value: zotero.url}) + "}}" });
    }
  }
```

producing:   note = {(accessed 2023-03-17T10:24:46Z)},

This only works as long as:
- I don't use the `if (zotero.accessDate)`.
- I don't use `replace("...", "") on zotero.accessDate 
Am I missing something?

Software:
Zotero Zotero 6.0.23
Ubuntu 20.04

Thanks in advance!

**Comment by retorquere** [comment, marked as answer]
I need a debug log for this entry.

**Reply by KevinYanesG** [reply]
Where can I find the debug logs? Or how exactly is the procedure to debug? 

Because I tried to use the `Zotero.debug(...)` method, but I couldn't find where the logs were stored/outputted.

**Reply by retorquere** [reply]
Right click the entry, and choose better bibtex - send debug log from the popup menu.

**Reply by KevinYanesG** [reply]
Ok, now it works, the behavior is the following:
- If I copy [this](https://retorque.re/zotero-better-bibtex/exporting/scripting/#add-accessdate-url-for-bibtex) refresh the bibTex and then change the `Translator` to be BibLaTex and refresh again it works.
- If I, afterwards, want to add something else like `Zotero.debug(...)`, then it doesn't work  anymore. If I then delete the `Zotero.debug(...)` lineand come back to the code that _was_ working, suddenly it doesn't work anymore. 
Looking at the logs it seems as if the Zotero.debug() would be still in cache although it is not anymore in the code:
```
const result = (() => { if (Translator.BetterBibLaTeX && zotero.itemType === 'webpage') {
    if (zotero.acessDate) {
      tex.add({ name: 'note', value: "(accessed " + zotero.accessDate.replace(/\s*T?\d+:\d+:\d+.*/, '') + ")" });
    }
  }; })();

    // entry type change through legacy field
    if (entry.entrytype === entrytype && reference.referencetype !== entrytype) entry.entrytype = reference.referencetype
    delete entry.referencetype

    switch (typeof result) {
      case 'undefined': return { cache: true, write: true }
      case 'boolean': return { cache: result, write: true }
      case 'object': return { cache: true, write: true, ...result }
      default:
        Zotero.debug('Unexpected postscript result ' + JSON.stringify(result));
        return { cache: false, write: true }
    }
```

- I need to copy the original code again and start over.
But, I guess my problem is solved, thanks!

**Reply by retorquere** [reply]
Still need a debug log if you need my help.

**Reply by KevinYanesG** [reply]
All good, I think my problem is solved for now. Thank you!

**Reply by KevinYanesG** [reply]
Another question though: is there a documentation of the attributes that the `zotero` objects have? Like for example: `zotero.accessDate` I didn't find anything.

**Reply by retorquere** [reply]
I've added the available fields to https://retorque.re/zotero-better-bibtex/exporting/scripting/#item-types-and-fields

**Reply by KevinYanesG** [reply]
Thank you so much!
